src/storage/write_ahead_log/disk_log_consumer_task.cpp

```diff
@@ -57,15 +58,6 @@ void DiskLogConsumerTask::DiskLogConsumerTaskLoop() {
   // input for this operating unit
   uint64_t num_bytes = 0, num_buffers = 0;
 
-  bool logging_metrics_enabled =
-      common::thread_context.metrics_store_ != nullptr &&
-      common::thread_context.metrics_store_->ComponentToRecord(metrics::MetricsComponent::LOGGING);
-
-  if (logging_metrics_enabled) {
-    // start the operating unit resource tracker
-    common::thread_context.resource_tracker_.Start();
-  }
-
```

Comment:
I've changed the metrics recording in this file. Maybe just remove the changes here to avoid the conflicts. Then I'll just submit my new changes. Otherwise looks good!

Reply:
I'm not sure it works if I move it back. It's technically a bug that we currently check outside of the loop if metrics are enabled since it can be toggled off at any time.

Reply:
You're correct. I've changed that in my branch as well but I guess it makes sense to fix the bug first here. I can resolve the conflicts later.